org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/concurrent/ReadRequest.java

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2019 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ide.server.concurrent;
+
+import java.util.concurrent.ExecutorService;
+
+import org.apache.log4j.Logger;
+import org.eclipse.xtext.util.CancelIndicator;
+import org.eclipse.xtext.xbase.lib.Functions.Function1;
+
+/**
+ * @author kosyakov - Initial contribution and API
+ * @since 2.11
+ */
+public class ReadRequest<V> extends AbstractRequest<V> {
+	private static final Logger LOG = Logger.getLogger(ReadRequest.class);
+
+	private final Function1<? super CancelIndicator, ? extends V> cancellable;
+
+	private final ExecutorService executor;
+
+	public ReadRequest(RequestManager requestManager, Function1<? super CancelIndicator, ? extends V> cancellable,
+			ExecutorService executor) {
+		super(requestManager);
+		this.cancellable = cancellable;
+		this.executor = executor;
+	}
+
+	@Override
+	public void run() {
+		if (result.isCancelled()) {
+			return;
+		}
+		this.executor.submit(() -> {
+			try {
+				cancelIndicator.checkCanceled();
+				result.complete(cancellable.apply(cancelIndicator));
+			} catch (Throwable t) {
+				if (!requestManager.isCancelException(t)) {
```

Comment:
This condition is not in the original code. Intentional or debug leftover?

Reply:
I see the description in the original code:
```
if (((e != null) && (!this.requestManager.isCancelException(e)))) {
```

Reply:
Interesting. I compared to my local code (pulled yesterday) and there the code is defnitly not there. Probably a statement that was introduced since yesterday and I'm a commit or two behind HEAD.

Reply:
Indeed. Another PR with added logging was merged before I submitted this one here on top of the updated code.